### PR TITLE
fix(flat-table-cell, flat-table-header): raise specificity of spacing styles FE-3838

### LINKED ISSUE #3846: Styled components padding does not apply correctly to table cells
### Current behaviour

As you can see below, the following code is used on a TableCell

`<FlatTableCell p={2}>`

Screenshot 2021-03-22 at 09 20 33

However the problem is the padding left and right is using the base table cell padding and not the provided override, whereas the cell has no top/bottom default padding therefore those worked correctly.

### Expected behaviour

The passed padding values should take precedence.

### PR DESCRIPTION
**Pull request by edleeks87**
fix #3846

### Proposed behaviour

Raises specificity of the `space` styles applied to div container within `FlatTableCell`, `FlatTableRowHeader` and
`FlatTableHeader` so any values passed via `styled-system` props take precedence of the cell-sizes
configs

Adds `width` prop for `FlatTableCell`. Adds `truncate` and `title` props to support styling content
that overflows the width of the column and overriding the text displayed when the mouse hovers over
a cell.

After speaking to @harpalsingh it was decided to remove the `nowrap` styling on cells and only apply it when `truncate` is true

### Current behaviour

Styled-system props are overridden by the size config
No ability to truncate cell content

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

### Testing instructions

`design-system-flat-table--with-truncated-cell-content` has been created

https://codesandbox.io/s/carbon-quickstart-forked-fm951?file=/src/index.js

### CODE REVIEW

File: src/components/flat-table/flat-table.stories.mdx [hunk 1]
```diff
@@ -208,6 +208,35 @@ Column width can be set by passing number as a `width` prop to the column header
   </Story>
 </Preview>
 
+### With tuncated cell content
+When setting column widths it is also possible to set the content to `truncate`: this will prevent the content from wrapping 
+and add ellipsis to any content that overflows the given width. By default the `title` will be set to the children if it 
+is a string. However, if the passed children are not a string it is possible to override it by passing in a differnt
```
Comment (DipperTheDan) [inline src/components/flat-table/flat-table.stories.mdx:214]: Little spelling mistake. `differnt` should be `different`.

File: src/components/flat-table/flat-table.stories.mdx [hunk 2]
```diff
@@ -208,6 +208,35 @@ Column width can be set by passing number as a `width` prop to the column header
   </Story>
 </Preview>
 
+### With tuncated cell content
+When setting column widths it is also possible to set the content to `truncate`: this will prevent the content from wrapping 
+and add ellipsis to any content that overflows the given width. By default the `title` will be set to the children if it 
+is a string. However, if the passed children are not a string it is possible to override it by passing in a differnt 
+`title` sting.
```
Comment (DipperTheDan) [inline src/components/flat-table/flat-table.stories.mdx:215]: Little spelling mistake here. `sting` should be `string`.

Comment (edleeks87) [reply src/components/flat-table/flat-table.stories.mdx:215]: Call `The Police`

File: src/components/flat-table/flat-table.stories.mdx [hunk 3]
```diff
@@ -208,6 +208,35 @@ Column width can be set by passing number as a `width` prop to the column header
   </Story>
 </Preview>
 
+### With tuncated cell content
```
Comment (bulaj) [inline src/components/flat-table/flat-table.stories.mdx:211]: typo

**Review by bulaj — APPROVED** [review]

**Comment by nicktitchmarsh** [comment]
Combining expandable rows with truncate doesn't actually truncate the content of the main row. Seems to work for sub rows however.

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by bulaj — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**